app/src/main/java/org/eyeseetea/malariacare/layout/adapters/survey/DynamicTabAdapter.java

```diff
@@ -820,10 +822,17 @@ private boolean toggleChild(TableRow row, Question rowQuestion, Question childQu
             Survey survey = SurveyFragmentStrategy.getSessionSurveyByQuestion(rowQuestion);
 
             if (rowQuestion.isHiddenBySurveyAndHeader(survey)) {
+                row.clearFocus();
                 row.setVisibility(View.GONE);
+                ((CommonQuestionView) row.getChildAt(0)).setActive(false);
+                Object inputView = ((AKeyboardQuestionView) row.getChildAt(0)).getInputView();
```

Comment:
I dont´t like this code, if a question is not active then inside it should remove input error, seems this responsibility is of questionview not dynamictabadapter